### ISSUE TITLE
Forbid string concat with anything than other strings

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#648]
# Rule request

## Thesis

These are fine:

```python
template1 = (  # multiline string
    'a' +
    'b' + 
    'c'
)

template2 = (  # multiline byte-string
    b'a' + 
    b'b' + 
    b'c'
)
```

These are not fine:

```python
a = 'a' + 'b'  # why not 'ab'?
b = b'b' + 'a'  # string + bytes
c = 'a' + variable  # should be 'a{0}'.format(variable)
```

## Reasoning

We should enforce consistency in string formatting:
- We already forbid `%`
- We forbid `f''`
- We forbid implicit concat: `'a' 'b'`
- Now we need to forbid explicit concatenation 

And the only way for string to format will be `.format()`
We also allow to use multi-line strings, since it is quite common and useful way to write long strings in your code.

**Comment by sobolevn** [comment]
We still allow `b'b' + 'a'`, since it did not feel right to add this check to this rule.
Moreover, it feels like type-checking that is out of scope.
